csrc/assume.cpp

```diff
@@ -0,0 +1,35 @@
+#include <assume.h>
+#include <ir_all_nodes.h>
+#include <ir_builder.h>
+
+#include <vector>
+
+namespace nvfuser::assume {
+
+Bool* tensorsAreNotEmpty(Val* value) {
+  std::vector<Val*> todo{value};
+  std::vector<Val*> tensor_sizes;
+  while (!todo.empty()) {
+    auto v = todo.back();
+    todo.pop_back();
+    if (auto ns = dynamic_cast<NamedScalar*>(v)) {
+      if (ns->isTensorSize()) {
+        tensor_sizes.emplace_back(v);
+        continue;
+      }
+    }
+    if (auto def = v->definition()) {
+      for (auto inp : def->inputs()) {
+        todo.emplace_back(inp);
+      }
+    }
+  }
+  Bool* result = nullptr;
+  for (auto ts : tensor_sizes) {
```

Comment:
`tensor_sizes` can contain tensor size named scalars redundantly. Are you assuming the redundancy should be eliminated by the expr simplification?

Reply:
Thanks for catching! They will not be eliminated by expr simplifier, because the `assumptions` argument are not simplified. I don't think this will cause any issue, but still, we should get rid of it. I will update this PR.

Reply:
fixed